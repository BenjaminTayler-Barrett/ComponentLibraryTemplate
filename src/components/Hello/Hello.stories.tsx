import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Hello from "./Hello";

export default {
    title: "Components/Hello",
    component: Hello,
} as ComponentMeta<typeof Hello>;

const Template: ComponentStory<typeof Hello> = (args) => <Hello {...args} />;

export const HelloStory = Template.bind({});
HelloStory.args = {
    text: "Hello World",
};
