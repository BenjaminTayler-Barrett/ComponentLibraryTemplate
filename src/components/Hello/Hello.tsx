import React from "react";
import "../../tailwind.css";
import "./Hello.scss";

export interface HelloProps {
    text: string;
}

class Hello extends React.Component<HelloProps> {
    override render() {
        return <div className="Hello text-3xl font-bold underline">{this.props.text}</div>;
    }
}

export default Hello;
