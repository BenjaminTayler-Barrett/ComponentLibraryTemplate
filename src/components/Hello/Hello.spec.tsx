import React from "react";
import { render, screen } from "@testing-library/react";
import Hello from "./Hello";

describe("Hello component", () => {
    it("should render Hello component", () => {
        const expectedText = "Hello World";

        const { container } = render(<Hello text={expectedText} />);

        expect(container).toMatchSnapshot();
    });

    it("should contain the correct text", () => {
        const expectedText = "Hello World";

        render(<Hello text={expectedText} />);

        expect(screen.getByText(expectedText)).toBeTruthy();
    });
});
