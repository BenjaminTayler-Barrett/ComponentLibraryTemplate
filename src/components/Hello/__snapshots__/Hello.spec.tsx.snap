// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component should render Hello component 1`] = `
<div>
  <div
    class="Hello text-3xl font-bold underline"
  >
    Hello World
  </div>
</div>
`;
